src/tables/system_logs/system_logs.windows.cc

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) 2021 by the Zeek Project. See LICENSE for details.
+//
+// Interface to journald. To avoid dependencies on external libraries, we spawn
+// journalctl as a child process if we find it, reading from its output.
+
+#include "system_logs.h"
+
+#include "core/database.h"
+#include "core/logger.h"
+#include "util/fmt.h"
+#include "util/windows-util.h"
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+using namespace zeek::agent;
+using namespace zeek::agent::table;
+
+namespace {
+
+// See the comment in getLogs() for info about this.
+constexpr int MAX_RECORDS_TO_READ = 50;
+
+enum class LogKind { System, Security };
+
+static std::string kind_string(LogKind kind) {
+    switch ( kind ) {
+        case LogKind::System: return "System";
+        case LogKind::Security: return "Security";
+        default: return "Unknown";
+    }
+}
+
+static std::wstring kind_wstring(LogKind kind) {
+    switch ( kind ) {
+        case LogKind::System: return L"System";
+        case LogKind::Security: return L"Security";
+        default: return L"Unknown";
+    }
+}
+
+struct LogEntry {
+    int64_t id;
+    int64_t ts;
+    LogKind kind;
+    std::string priority;
+    std::string source;
+    std::string message;
+};
+
+struct LogHandle {
+    HANDLE handle;
+    DWORD last_read;
+    LogKind kind;
+};
+
+class SystemLogsWindows : public SystemLogs {
+public:
+    void activate() override;
+    void deactivate() override;
+    void poll() override;
+
+private:
+    void getLogs(LogHandle& log_handle, std::vector<LogEntry>& results);
+    std::optional<LogEntry> processRecord(char* buffer, PEVENTLOGRECORD record, LogKind kind);
+
+    LogHandle system{nullptr, 0, LogKind::System};
+    LogHandle security{nullptr, 0, LogKind::Security};
+
+    std::map<std::wstring, HMODULE> dll_cache;
+};
+
+database::RegisterTable<SystemLogsWindows> _;
+
+void SystemLogsWindows::activate() {
+    system.handle = OpenEventLog(NULL, "System");
+    if ( ! system.handle ) {
+        std::error_condition cond = std::system_category().default_error_condition(GetLastError());
+        logger()->info(format("Failed to open System event log: {}", cond.message()));
+    }
+
+    // The system log requires elevated privileges to open. Right now, it'll just log something if it
```

Comment:
You can silence this message, turn it into debug message I'd say. Other tables often need admin, too, to get everything and the agent reports a general "information may be missing " message when not running as admin.

Reply:
I moved the other one to debug as well for consistency.